x/bank/keeper/keeper.go

```diff
@@ -454,8 +459,9 @@ func (k BaseViewKeeper) IterateTotalSupply(ctx sdk.Context, cb func(sdk.Coin) bo
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		amount, ok := sdk.NewIntFromString(string(iterator.Value()))
-		if !ok {
+		var amount sdk.Int
+		err := amount.Unmarshal(iterator.Value())
+		if err != nil {
 			panic("unexpected supply")
```

Comment:
same as above: I would put the error here as `fmt.Errorf("unexpected supply amount: %w", err)`